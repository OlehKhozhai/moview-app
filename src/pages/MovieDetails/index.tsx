import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

import Footer from 'components/Footer';
import MovieDetailsBanner from 'components/MovieDetailsBanner';
import Divider from 'components/_common/Divider';
import MoviesNavbar from 'components/MoviesNavbar';
import MoviesList from 'components/MoviesList';
import { getMovieDetailsAction, getMoviesAction } from 'redux/actions';
import { movieDetailsSelector, moviesSelector } from 'redux/selectors';
import useSearchParams from 'hooks/useSearchParams';

const MovieDetails = () => {
  const dispatch = useDispatch();
  const movieDetails = useSelector(movieDetailsSelector);
  const { movies, isMoviesLoading } = useSelector(moviesSelector);
  const { id: movieId } = useParams<{ id: string }>();
  const searchParams = useSearchParams();

  React.useEffect(() => {
    if (movieId) {
      dispatch(getMovieDetailsAction(+movieId));
    }
  }, [movieId, dispatch]);

  React.useEffect(() => {
    if (searchParams) {
      dispatch(getMoviesAction({ params: searchParams }));
    }
  }, [searchParams, dispatch]);

  return (
    <>
      {movieDetails && <MovieDetailsBanner movie={movieDetails} />}
      <Divider />
      <MoviesNavbar searchParams={searchParams} />
      <MoviesList movies={movies} isLoading={isMoviesLoading} />
      <Footer />
    </>
  );
};

export default React.memo(MovieDetails);
