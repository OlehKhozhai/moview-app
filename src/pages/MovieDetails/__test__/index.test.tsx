import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import MovieDetails from '../index';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest
    .fn()
    .mockReturnValue({ movies: [], isMoviesLoading: false, movie: { genres: [] } }),
}));

jest.mock('components/Footer/', () => ({
  __esModule: true,
  default: () => '---Footer---',
}));
jest.mock('components/MovieDetailsBanner/', () => ({
  __esModule: true,
  default: () => '---MovieDetailsBanner---',
}));
jest.mock('components/_common/Divider', () => ({
  __esModule: true,
  default: () => '---Divider---',
}));
jest.mock('components/MoviesNavbar/', () => ({
  __esModule: true,
  default: () => '---MoviesNavbar---',
}));
jest.mock('components/MoviesList/', () => ({
  __esModule: true,
  default: () => '---MoviesList---',
}));

describe('MovieDetails component', () => {
  test('should render markup by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <MovieDetails />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
