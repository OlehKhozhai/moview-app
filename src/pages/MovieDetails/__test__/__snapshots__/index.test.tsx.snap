// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component should render markup by default 1`] = `
Array [
  "---MovieDetailsBanner---",
  "---Divider---",
  "---MoviesNavbar---",
  "---MoviesList---",
  "---Footer---",
]
`;
